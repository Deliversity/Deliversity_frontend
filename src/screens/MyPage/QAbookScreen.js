import React, {Component, useState} from 'react';
import {View, StyleSheet, RefreshControl, FlatList} from 'react-native';
import axios from '../../axiosConfig';
import Card from '../../components/QAcard';
import {Button, Text} from 'native-base';
class QAbookScreen extends Component {
  static navigationOptions = {
    title: 'QAbook',
  };
  constructor(props) {
    super(props);

    this.state = {
      content: '',
      refreshing: false,
    };
  }
  componentDidMount(): void {
    this.getQAList();
  }
  onRefresh = async () => {
    await this.getQAList();
  };
  getQAList = async () => {
    try {
      this.setState({refreshing: true});
      await axios
        .get('/api/v1/myinfo/qnas')
        .then((res) => {
          this.setState({
            content: res.data.data,
            refreshing: false,
          });
        })
        .catch((error) => {
          alert(error.response.data.message);
        });
    } catch (e) {
      alert('error' + e);
    }
  };
  renderItem = ({item}) => {
    return <Card itemData={item} />;
  };

  render() {
    return (
      <View style={styles.container}>
        <View style={styles.footer}>
          <View style={styles.write}>
            <FlatList
              refreshControl={
                <RefreshControl
                  refreshing={this.state.refreshing}
                  onRefresh={this.onRefresh}
                />
              }
              extraData={this.state}
              data={this.state.content}
              renderItem={this.renderItem}
              keyExtractor={(item) => item.id.toString()}
            />
          </View>
        </View>
        <View
          style={{
            backgroundColor: '#fff',
            flex: 1,
            justifyContent: 'flex-end',
            flexDirection: 'row',
          }}>
          <Button
            rounded
            dark
            style={{marginRight: 15}}
            onPress={() => {
              this.props.navigation.navigate('QApage');
            }}>
            <Text style={{color: '#fff'}}>문의하러가기</Text>
          </Button>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  footer: {
    flex: 8,
    backgroundColor: '#fff',
    paddingVertical: 20,
    paddingHorizontal: 10,
    flexDirection: 'column',
  },
  write: {
    marginTop: 10,
  },
});

export default QAbookScreen;
