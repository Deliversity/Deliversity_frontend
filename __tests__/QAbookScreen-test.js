import 'react-native';

import React, {ReactElement} from 'react';
import QAbookScreen from '../src/screens/MyPage/QAbookScreen';
import renderer from 'react-test-renderer';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
configure({adapter: new Adapter()});
let props;
let component: ReactElement;

beforeAll(() => {
  jest.mock('@react-native-community/async-storage');
});

describe('[Temp] render', () => {
  props = {}; // fill test props
  const params = {};
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<QAbookScreen {...props} />);
    wrapper.setState({params: params});
  });
  it('renders without crashing', () => {
    // const renderResult = renderer.create(wrapper);
    // const renderedJson = wrapper.toJSON();
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(toJson(wrapper)).toBeTruthy();
  });
});
