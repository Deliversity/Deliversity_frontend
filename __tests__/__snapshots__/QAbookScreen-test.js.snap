// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Temp] render renders without crashing 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 8,
        "flexDirection": "column",
        "paddingHorizontal": 10,
        "paddingVertical": 20,
      }
    }
  >
    <View
      style={
        Object {
          "marginTop": 10,
        }
      }
    >
      <FlatList
        data=""
        disableVirtualization={false}
        extraData={
          Object {
            "content": "",
            "params": Object {},
            "refreshing": true,
          }
        }
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onEndReachedThreshold={2}
        refreshControl={
          <RefreshControlMock
            onRefresh={[Function]}
            refreshing={true}
          />
        }
        removeClippedSubviews={true}
        renderItem={[Function]}
        scrollEventThrottle={50}
        updateCellsBatchingPeriod={50}
        windowSize={21}
      />
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-end",
      }
    }
  >
    <Styled(Button)
      dark={true}
      onPress={[Function]}
      rounded={true}
      style={
        Object {
          "marginRight": 15,
        }
      }
    >
      <Styled(Text)
        style={
          Object {
            "color": "#fff",
          }
        }
      >
        문의하러가기
      </Styled(Text)>
    </Styled(Button)>
  </View>
</View>
`;
